deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/config/ZKMysqlDataSourceFactory.java

```diff
@@ -84,9 +93,47 @@ public BasicDataSource build() throws Exception {
             host = reader.getHost(replicaSet);
             port = reader.getPort(replicaSet);
         }
-        KnoxDBKeyReader.init((role));
-        String userName = KnoxDBKeyReader.getUserName();
-        String password = KnoxDBKeyReader.getPassword();
-        return DatabaseUtil.createMysqlDataSource(host, port, userName, password, pool);
+
+        if (this.useMTLS) {
+            String userName = getUserNameFromSpiffeId();
+            String password = "password";
+            Map<String, String> proxyConnectionProps = ImmutableMap.<String, String>builder()
+                // ssl properties
+                .put("sslMode", "VERIFY_CA" )
+                .put("trustCertificateKeyStoreUrl", "file:/var/lib/normandie/fuse/jkstrust/generic" )
+                .put("trustCertificateKeyStoreType", "JKS" )
+                .put("trustCertificateKeyStorePassword", "pintastic" )
```

Comment:
Do you know if this pintastic password (and the one a few lines below) is supposed to be the same for every db connection? Have you tried a different value?